server/src/immich/api-v1/asset/asset.service.ts

```diff
@@ -369,4 +375,8 @@ export class AssetService {
       throw new BadRequestException('Quota has been exceeded!');
     }
   }
+
+  private async fixModifiedAtTimestamp(file: UploadFile, dto: CreateAssetDto) {
+    await utimes(file.originalPath, new Date(), new Date(dto.fileModifiedAt));
```

Comment:
We don't call fs methods directly in services. Can you just add this to the storage repository and use that here instead? It makes the tests simpler.

Reply:
Sounds good, I've added a `utimes` method to the storage repository/filesystem provider